src/oic/oic/provider.py

```diff
@@ -1681,17 +1682,24 @@ def provider_features(self, pcr_class=ProviderConfigurationResponse):
 
         _provider_info = pcr_class(**CAPABILITIES)
 
+        # Parse scopes
+        _scopes = list(SCOPE2CLAIMS.keys())
+        if self.extra_scope_dict is not None:
+            _scopes.extend(self.extra_scope_dict.keys())
+        # Remove duplicates if any
+        _provider_info["scopes_supported"] = list(set(_scopes))
```

Comment:
You can actually write this shorter and without a copy...

_provider_info[...] = list(set(itertools.chain(SCOPE2CLAIMS.keys(), self.extra_scope_dict.keys() if self.extra_scope_dict else []))

but i think your version is more readable ;-)

Reply:
I will stick to mine, I guess readability is more important... :)